hmda-data-publisher/src/main/scala/hmda/publisher/scheduler/TsScheduler.scala

```diff
@@ -139,222 +139,82 @@ class TsScheduler(publishingReporter: ActorRef[PublishingReporter.Command], qaFi
-        }
-      }
-    case TsScheduler2021 =>
-      publishingGuard.runIfDataIsValid(Period.y2021, Scope.Private) {
-        val now           = LocalDateTime.now().minusDays(1)
-        val formattedDate = fullDate.format(now)
-        val fileName      = s"$formattedDate" + "2021_ts.txt"
-        val s3Path        = s"$environmentPrivate/ts/"
-        val fullFilePath  = SnapshotCheck.pathSelector(s3Path, fileName)
-
-        def countF: Future[Int] = tsRepository2021.count()
-
-        val results = for {
-          count <- countF
-          s3Sink = S3
-            .multipartUpload(bucketPrivate, fullFilePath, metaHeaders = MetaHeaders(Map(LarScheduler.entriesCountMetaName -> count.toString)))
-            .withAttributes(S3Attributes.settings(s3Settings))
-          _ <- uploadFileToS3(s3Sink, tsRepository2021.getAllSheets(getFilterList()))
-          _ <- persistFileForQa(fullFilePath, qaTsRepository2021)
-          count <- countF
-        } yield count
-
-        results onComplete {
-          case Success(count) => reportPublishingResult(TsScheduler2021, fullFilePath,Some(count))
-          case Failure(t) => reportPublishingResultError(TsScheduler2021, fullFilePath,t)
-        }
-      }
+    case TsScheduler2020 => publishAnnualTsData(TsScheduler2020, Period.y2020, "2020_ts.txt", tsRepository2020, qaTsRepository2020)
+
+    case TsScheduler2021 => publishAnnualTsData(TsScheduler2021, Period.y2021, "2021_ts.txt", tsRepository2021, qaTsRepository2021)
 
     case TsSchedulerQuarterly2020 =>
-      val now           = LocalDateTime.now().minusDays(1)
-      val formattedDate = fullDateQuarterly.format(now)
-      val s3Path        = s"$environmentPrivate/ts/"
-      def publishQuarter[Table <: RealTransmittalSheetTable2020](
-                                                              quarter: Period.Quarter,
-                                                              repo: TSRepository2020Base[Table],
-                                                              fileNameSuffix: String,
-                                                              qaRepository: QARepository[TransmittalSheetEntity]
-                                                            ) =
-        timeBarrier.runIfStillRelevant(quarter) {
-          publishingGuard.runIfDataIsValid(quarter, Scope.Private) {
-            val fileName     = formattedDate + fileNameSuffix
-            val fullFilePath = SnapshotCheck.pathSelector(s3Path, fileName)
-
-            def countF: Future[Int] = repo.count()
-
-            val results = for {
-              count <- countF
-              s3Sink = S3
-                .multipartUpload(bucketPrivate, fullFilePath, metaHeaders = MetaHeaders(Map(LarScheduler.entriesCountMetaName -> count.toString)))
-                .withAttributes(S3Attributes.settings(s3Settings))
-              _ <- uploadFileToS3(s3Sink, repo.getAllSheets(getFilterList()))
-              count <- countF
```

Comment:
for yearly data `persistFileForQa()` which takes in the `qaRepository` is called, but not for quarterly even though the constructed `publishQuarter()` method takes in that parameter, I'm wondering if this is done on purpose, or oversight which we should rectify and `persistFileQa()`

Reply:
@lchen-2101  this was most likely an oversight